dvc/remote/gs.py

```diff
@@ -22,33 +25,49 @@ def wrapper(*args, **kwargs):
         import requests
         from google.cloud.storage.blob import Blob, _DEFAULT_CHUNKSIZE
 
-        # Default chunk size for gs is 100M, which might be too much for
-        # particular network (see [1]). So if we are getting ConnectionError,
-        # we should try lowering the chunk size until we reach the minimum
-        # allowed chunk size of 256K. Also note that `chunk_size` must be a
-        # multiple of 256K per the API specification.
+        # `ConnectionError` may be due to too large `chunk_size`
+        # (see [#2572]) so try halving on error.
+        # Note: default 100M is too large for fine-grained progress
+        # so 10M is the starting default.
+        # Note: minimum 256K.
+        # Note: must be multiple of 256K.
         #
-        # [1] https://github.com/iterative/dvc/issues/2572
+        # [#2572]: https://github.com/iterative/dvc/issues/2572
 
         # skipcq: PYL-W0212
-        multiplier = int(_DEFAULT_CHUNKSIZE / Blob._CHUNK_SIZE_MULTIPLE)
+        multiplier = int(
+            _DEFAULT_CHUNKSIZE / (10.0 * Blob._CHUNK_SIZE_MULTIPLE)
+        )
```

Comment:
hope this reduction 100M => 10M is fine. I really doubt it'll slow uploads but it definitely adds nice progress.

Reply:
Just set the multiplier to ~40(for 10M), no need to modify _DEFAULT_CHUNKSIZE from gs :)

Reply:
btw the `download_to_file` function doesn't have a chunk argument and seems to be doing something closer to 256K based on how frequently progress is updated.

Reply:
@casperdcl The chunk size there is created on blob initialization, and when it is not set, it is using https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/blob.py#L634
```
from google.resumable_media.requests import Download                            
```
https://github.com/googleapis/google-resumable-media-python/blob/master/google/resumable_media/requests/download.py#L49
which is using SINGLE_GET_CHUNK_SIZE from https://github.com/googleapis/google-resumable-media-python/blob/master/google/resumable_media/requests/_helpers.py#L28 , which is ~8KB O_o

Reply:
yeah well AFAIK download should have bigger chunk size than uploads (since in general donwload speeds are greater than uploads).

To me 8k sounds tiny and 10M sounds massive.

Personally would prefer 512k for upload and 1M for download.